R/cube-transforms.R

```diff
@@ -0,0 +1,177 @@
+#' Show the variable transformations on a CrunchCube
+#'
+#' @param x a CrunchCube
+#'
+#' @return an array with any transformations applied
+#'
+setMethod("showTransforms", "CrunchCube", function (x) {
+    if (is.null(transforms(x))) {
+        print(cubeToArray(x))
+        return(invisible(x))
+    } else {
+        appliedTrans <- applyTransforms(x)
+        row_cats <- Categories(data=index(variables(x))[[1]]$categories)
+        row_styles <- transformStyles(transforms(x), row_cats[!is.na(row_cats)])
+
+        if (length(dim(appliedTrans)) <= 2) {
+            out <- prettyPrint2d(appliedTrans, row_styles = row_styles)
+            cat(unlist(out), sep="\n")
+        } else {
+            # styling is hard
+            print(appliedTrans)
+        }
+
+        return(invisible(appliedTrans))
+    }
+})
+
+#' @rdname applyTransforms
+#' @export
+setMethod("subtotalArray", "CrunchCube", function(x, headings = FALSE) {
+    includes <- c("subtotals")
+    if (headings) {
+        includes <- c(includes, "headings")
+    }
+    applyTransforms(x, include = includes)
+})
+
+#' Calculate the transforms from a CrunchCube
+#'
+#' `applyTransforms` calculates transforms (e.g. subtotals) on a CrunchCube.
+#' Currently only the row transforms are supported. This is useful if you want
+#' to use the values from the subtotals of the CrunchCube in an analysis.
+#'
+#' Including the `include` argument allows you to specify which parts of the
+#' CrunchCube to return. The options can be any of the following: "cube_cells"
+#' for the untransformed values from the cube itself, "subtotals" for the
+#' subtotal insertions, and "headings" for any additional headings. Any
+#' combination of these can be given, by default all will be given.
+#'
+#' `subtotalArray(cube)` is a convenience function that is equivalent to
+#' `applyTransforms(cube, include = c("subtotals"))`
+#'
+#' @param x a CrunchCube
+#' @param ary an array to use, if not using the default array from the cube
```

Comment:
`ary` is weird and no one will remember how to (mis)spell that to use it. Can you pick something better?